src/client/dfs/dfs.c

```diff
@@ -2238,6 +2237,10 @@ dfs_remove(dfs_t *dfs, dfs_obj_t *parent, const char *name, bool force,
 	if (!exists)
 		D_GOTO(out, rc = ENOENT);
 
+	/* Check if the file is the expected one */
+	if ((oid && (oid->lo || oid->hi)) && (oid->lo != entry.oid.lo || oid->hi != entry.oid.hi))
```

Comment:
why check  (oid->lo || oid->hi) ?

Reply:
I think you need a different variant here. maybe dfs_remove_internal() with a flag that indicates whether the oid is passed in to check (dfs_removeoi) or returned to user (dfs_remove)

Reply:
yes, checking oid->lo and oid->hi is to check if this is intended to be a input or input/output parameter.  At extra flag is probably needed to make this clearer.

Reply:
I backout out all the changed to dfs_remove() for now.   My intention here was to add and test for the new features that dfuse needs from dfs for multi-user but I ended up realising that dfuse could more with the oids as outputs than it was, so this PR only adds that, and does not yet attempt to provide oids as inputs at all.